internal/bminventory/inventory.go

```diff
@@ -1003,6 +1003,14 @@ func (b *bareMetalInventory) GenerateClusterISO(ctx context.Context, params inst
 func (b *bareMetalInventory) GenerateClusterISOInternal(ctx context.Context, params installer.GenerateClusterISOParams) (*common.Cluster, error) {
 	log := logutil.FromContext(ctx, b.log)
 	log.Infof("prepare image for cluster %s", params.ClusterID)
+
+	if params.ImageCreateParams.SSHPublicKey != "" {
```

Comment:
If this parameter is not set didn't we talked about taking it from the cluster ssh-key? @avishayt

Reply:
I don't think so. Around line 1090 in the same function we call `updates["image_ssh_public_key"] = params.ImageCreateParams.SSHPublicKey`, no reference to cluster's SSH public key whatsoever . Actually, I opened a bug about a cluster's and image SSH public keys being completely independent. At least I couldn't find any relationship in the code and after running a few tests.

Reply:
And the image-level key makes it all the way into ignition https://github.com/empovit/assisted-service/blob/076a14a6e78d4f263abb0de6cd6ea9e37621ac5a/internal/bminventory/inventory.go#L397

Reply:
Wouldn't it be better to mark it as required in the swagger?

Reply:
No because it's not required.